hw6/queen.py

```diff
@@ -0,0 +1,38 @@
+from itertools import product, combinations
+
+
+def check(array_coord):
+    for i in range(len(array_coord)-1):
+        for j in range(i+1, len(array_coord)):
+            x1 = int(array_coord[i][0])
+            x2 = int(array_coord[j][0])
+            y1 = int(array_coord[i][1])
+            y2 = int(array_coord[j][1])
+            if x1 == x2 or y1 == y2 or abs(x1 - x2) == abs(y1 - y2):
+                return False
+    return True
```

Comment:
Можно сделать значительно проще.
```suggestion
    for i, (x1, y1) in enumerate(array_coord, 1):
        for x2, y2 in cords[i:]:
            if x1 == x2 or y1 == y2 or abs(x01 - x2) == abs(y1 - y2):
                return False

    return True
```